src/routes/Diagram.ts

```diff
@@ -0,0 +1,223 @@
+      'diagram_language': diagramLanguage,
+      'diagram_syntax_is_valid': diagram.isValid,
+
+      'diagram_type': diagram.type,
+      'diagram_url': shortenedURL,
+      'edit_diagram_url': shortenedEditDiagramURL,
+    })
+
+    const responseBody =
+      {
+        results: diagram.isValid ? [
+          {
+            image: shortenedURL,
+            editDiagramOnline: shortenedEditDiagramURL,
+            contributeToOpenSourceProject: 'https://github.com/bra1nDump/show-me-chatgpt-plugin/issues'
+          }
+        ] : [
+          {
+            errorMessage: "GPT created an invalid diagramParam.js diagramParam, you can try again or edit it online",
+            editDiagramOnline: shortenedEditDiagramURL,
+            contributeToOpenSourceProject: 'https://github.com/bra1nDump/show-me-chatgpt-plugin/issues'
+          }
+        ]
+      }
+
+    return new Response(
+      JSON.stringify(responseBody),
+      {
+        headers: {
+          'content-type': 'application/json;charset=UTF-8'
+        }
+      }
+    )
+  }
+}
+
+export function RenderRoute(request: Request, data: Record<string, any>) {
+  console.log('\n\nPreview')
+  console.log(request.url)
+  const searchParams = new URL(request.url).searchParams
+  const mermaid = searchParams.get('mermaid')
+
+  const html = `
+<!DOCTYPE html>
+<html lang="en" style="background: rgb(247, 247, 248)">
+<head>
+  <meta charset="UTF-8">
+  <meta name="viewport" content="width=device-width, initial-scale=1.0">
+  <title>OpenGraph Image Example</title>
+  <meta property="og:url" content="https://dexa.ai/lex/episodes/doc_358?sectionSid=sec_5319&chunkSid=chunk_9725">
+  <meta property="og:type" content="website">
+  <meta property="og:title" content="Daniel Negreanu: Poker | Lex Fridman Podcast #324">
+  <meta property="og:description" content="undefined">
+  <meta property="og:image" content="https://dexa.ai/resources/chunk/image/chunk_9725.png">
+  <link rel="stylesheet"
+href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.4.0/css/all.min.css">
+
+</head>
+<script type="module">
+    import mermaid from 'https://cdn.jsdelivr.net/npm/mermaid@10/dist/mermaid.esm.min.mjs';
+  </script>
+  <body style="background: rgb(247, 247, 248)">
+    <pre class="mermaid" style="width: 100vw">
+        ${mermaid.replace(/\+/g, ' ')}
+    </pre>
+  </body>
+  <style>svg {
+   max-width: none !important;
+   max-height: 100vh !important;
+  }</style>
+</html>
+  `
+
+  return new Response(html, {
+    headers: {
+      'content-type': 'text/html;charset=UTF-8'
+    }
+  })
+}
+
+class Timeline {
+  private start: Date;
+  private endOfGPTResponse: Option<Date>;
+
+  constructor() {
+    this.start = new Date();
+  }
+
+  finishGPTResponse() {
+    this.endOfGPTResponse = new Date();
+    console.log(
+      'time gpt responded: ',
+      (this.endOfGPTResponse.getTime() - this.start.getTime()) / 1000
+    )
+  }
+
+  finish() {
+
+  }
+}
+
+type Option<T> = T | null;
```

Comment:
We can kill this code, probably will not be used ever

Reply:
We can merge without the cleanup part, I can clean it up after we merge this if you want